CDCgov/prime-reportstream
# Deliveries: Filter Query by Creation Date

This PR adds date filtration to the Deliveries endpoint call through the Daily Data UI.

Test Steps:
1. Load the /daily-data page locally
2. Ensure your network call is going out with a query string attached to the URL
3. Since it's hard to test locally without days wort of data, here's an example on my local:

Unfiltered
<img width="789" alt="Screen Shot 2022-10-18 at 3 56 00 PM" src="https://user-images.githubusercontent.com/4022304/196531058-07373d34-a888-4106-9c97-378756b2525d.png">

Filtered
<img width="793" alt="Screen Shot 2022-10-18 at 3 55 48 PM" src="https://user-images.githubusercontent.com/4022304/196531009-7cb7c8f3-e0d7-437f-8666-58caa49574e3.png">

## Changes
- Adds TableFilters to Daily Data
- FilterManager tests added
- extractFiltersFromManager and additional tests added
- Cleaned up Daily Data JSX

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #4861 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
